File: msgr-user/src/main/java/ru/vsu/cs/msgr_user/dto/UpdateProfileRequest.java
package ru.vsu.cs.msgr_user.dto;

import jakarta.validation.constraints.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UpdateProfileRequest {

    @Size(max = 100, message = "Имя не может быть длиннее 100 символов")
    private String firstName;

    @Size(max = 100, message = "Фамилия не может быть длиннее 100 символов")
    private String lastName;

    @Size(min = 3, max = 50, message = "Username от 3 до 50 символов")
    @Pattern(regexp = "^[a-zA-Z0-9_]+$", message = "Username может содержать только латиницу, цифры и _")
    private String username;

    @Size(max = 500, message = "URL аватара не может быть длиннее 500 символов")
    private String avatarUrl;
}

File: messenger_app/lib/features/user/data/dto/update_profile_request.dart
class UpdateProfileRequest {
  final String? firstName;
  final String? lastName;
  final String? username;
  final String? avatarUrl;

  UpdateProfileRequest({
    this.firstName,
    this.lastName,
    this.username,
    this.avatarUrl,
  });

  Map<String, dynamic> toJson() => {
        if (firstName != null) 'firstName': firstName,
        if (lastName != null) 'lastName': lastName,
        if (username != null) 'username': username,
        if (avatarUrl != null) 'avatarUrl': avatarUrl,
      };
}
